File: app/src/main/java/org/antrack/app/ui/fragments/ScreensFragment.java
package org.antrack.app.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import org.antrack.app.ui.State;
import org.antrack.app.ui.U;

import app.R;

public class ScreensFragment extends ImagesFragment {
    @Override
    public String getMod() { return Mod.SCREENSHOT; }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (!State.features.root) {
            showNoRoot();
            return null;
        }

        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.fragment_screens, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case R.id.toolbar_action_front_camera:
                if (Mod.check(getMod())) {
                    U.runCommandAsync(getMod());
                } else {
                    Mod.showNoModule(getActivity(), getMod());
                }
                return true;
            case R.id.toolbar_action_delete:
                showRemoveDialog();
                return true;
        }
        return false;
    }
}

File: app/src/main/java/org/antrack/app/ui/fragments/ScreensFragment.kt
package org.antrack.app.ui.fragments

import android.os.Bundle
import android.view.*
import app.R
import org.antrack.app.ui.U

class ScreensFragment : ImagesFragment() {
    override val module = Mod.SCREENSHOT

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        checkRoot() || return null

        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.fragment_screens, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        when (id) {
            R.id.toolbar_action_front_camera -> {
                if (checkModule()) {
                    U.runCommandAsync(module)
                } else {
                    showNoModuleToast()
                }
                return true
            }
            R.id.toolbar_action_delete -> {
                showRemoveDialog()
                return true
            }
        }
        return false
    }
}
